web-app/js/portal/ui/openlayers/control/SpatialConstraint.js

```diff
@@ -13,7 +13,7 @@ Portal.ui.openlayers.SpatialConstraintType = {
 
```

Comment:
This class is quite big, and complex, IMO.  What are its main responsibilities currently (I could answer that but since you've just looked at the code, perhaps you're in a better position to do so)?